code/modules/mob/living/simple_animal/crow/crow.dm

```diff
@@ -38,101 +35,36 @@
-		var/obj/item/removed
-		switch(href_list["remove_inv"])
-			if("access cuff")
-				removed = access_card
-				access_card = null
-			if("back")
-				removed = messenger_bag
-				messenger_bag = null
-		if(removed)
-			removed.dropInto(loc)
-			usr.put_in_hands(removed)
-			visible_message("<span class='notice'>\The [usr] removes \the [removed] from \the [src]'s [href_list["remove_inv"]].</span>")
-			show_stripping_window(usr)
-			update_icon()
-		else
-			to_chat(user, "<span class='warning'>There is nothing to remove from \the [src]'s [href_list["remove_inv"]].</span>")
-		return TOPIC_HANDLED
-	if(href_list["add_inv"])
-		var/obj/item/equipping = user.get_active_hand()
-		if(!equipping)
-			to_chat(user, "<span class='warning'>You have nothing in your hand to put on \the [src]'s [href_list["add_inv"]].</span>")
-			return 0
-		var/obj/item/equipped
-		var/checktype
-		switch(href_list["add_inv"])
-			if("access cuff")
-				equipped = access_card
-				checktype = /obj/item/card/id
-			if("back")
-				equipped = messenger_bag
-				checktype = /obj/item/storage/messenger
-		if(equipped)
-			to_chat(user, "<span class='warning'>There is already something worn on \the [src]'s [href_list["add_inv"]].</span>")
-			return TOPIC_HANDLED
-		if(!istype(equipping, checktype))
-			to_chat(user, "<span class='warning'>\The [equipping] won't fit on \the [src]'s [href_list["add_inv"]].</span>")
-			return TOPIC_HANDLED
-		switch(href_list["add_inv"])
-			if("access cuff")
-				access_card = equipping
-			if("back")
-				messenger_bag = equipping
-		if(!user.try_unequip(equipping, src))
-			return TOPIC_HANDLED
-		visible_message("<span class='notice'>\The [user] places \the [equipping] on to \the [src]'s [href_list["add_inv"]].</span>")
-		update_icon()
-		show_stripping_window(user)
-		return TOPIC_HANDLED
+// Let people interact with the Bird Storage.
+/mob/living/simple_animal/crow/attack_hand(mob/user)
+	if(user.a_intent == I_HELP)
+		var/obj/item/backpack = get_equipped_item(slot_back_str)
+		if(backpack)
+			return backpack.attack_hand(user)
 	return ..()
 
-/mob/living/simple_animal/crow/show_examined_worn_held_items(mob/user, distance, infix, suffix, hideflags, decl/pronouns/pronouns)
-	. = ..()
-	if(Adjacent(src))
-		if(messenger_bag)
-			if(messenger_bag.contents.len)
-				to_chat(user, "It's wearing a little messenger bag with a Corvid Couriers logo on it. There's something stuffed inside.")
-			else
-				to_chat(user, "It's wearing a little messenger bag with a Corvid Couriers logo on it. It seems to be empty.")
-		if(access_card)
-			to_chat(user, "It has an access cuff with \the [access_card] inserted.")
+/mob/living/simple_animal/crow/attackby(obj/item/I, mob/user)
+	if(user.a_intent == I_HELP)
+		var/obj/item/backpack = get_equipped_item(slot_back_str)
+		if(backpack)
+			return backpack.attackby(I, user)
+	return ..()
 
 /mob/living/simple_animal/crow/on_update_icon()
 	..()
-	if(messenger_bag)
+	if(get_equipped_item(slot_back_str))
```

Comment:
Should it maybe check the type of the equipped item?

Reply:
As mentioned on Discord, would be good for this to use the animal bodytype and slot overlay systems now that they're available on non-humantypes. Will need extra care to avoid using the fallback human icons, though.